### DISCUSSION TITLE
Integer length in ParMETIS calls

### DISCUSSION THREAD

**Discussion by ivan-pi** [UvaCsl/HemoCell#51] [General discussion]
According to the [build script](https://github.com/UvaCsl/HemoCell/blob/master/external/setup.sh), HemoCell appears to use the default ParMETIS types `idx_t` and `real_t` in the [LoadBalancer](https://github.com/UvaCsl/HemoCell/blob/master/helper/loadBalancer.cpp). The types are defined one level higher in the METIS header `"metis.h"`, and are 32-bit types by default (i.e. `int32_t` and `float`).

Are the 32-bit types sufficient for HemoCell?

The integer type is signed, so the largest value representable is 2147483647.

**Comment by gzavo** [comment]
Hi @ivan-pi,

Thanks for pointing this out! The loadBalancer is expected to undergo heavy development this fall. We'll take a look at this then.

Bests,
Gabor

**Comment by ivan-pi** [comment]
While investigating partitioners, I noticed the Dolfinx developers have a common interface to three partitioners (Scotch, ParMETIS, KaHIP): https://github.com/FEniCS/dolfinx/blob/c2d1e71f1f9ac84bbf4c1fa81a8305f3ad043249/cpp/dolfinx/graph/partitioners.h

Perhaps useful as reference.

**Reply by gzavo** [reply]
It is indeed, thanks!
